app/src/main/java/space/taran/arkmemo/files/text/TextNoteFiles.kt

```diff
@@ -2,57 +2,69 @@ package space.taran.arkmemo.files.text
 
 import android.content.Context
 import android.util.Log
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
 import space.taran.arkmemo.files.parsers.JsonParser
 import space.taran.arkmemo.models.TextNote
 import space.taran.arkmemo.preferences.MemoPreferences
 import java.io.*
+import java.nio.charset.Charset
 import java.nio.file.Files
 import java.nio.file.Path
+import java.security.MessageDigest
+import kotlin.io.path.Path
 import kotlin.io.path.extension
 
-class TextFiles {
+class TextNoteFiles {
     companion object{
         private const val NOTE_EXT = "note"
-        private const val FILENAME = "Note"
 
-        private fun createFile(path: Path?, noteString: String?){
-            var numberOfFiles = 0
-            if(path != null) {
-                Files.list(path).forEach {
-                    if (it.fileName.extension == NOTE_EXT)
-                        numberOfFiles += 1
+        private fun createTextNoteFile(path: Path?, noteString: String?, filename: String){
+            fun writeToFile(bufferedWriter: BufferedWriter){
+                with(bufferedWriter) {
+                    write(noteString)
+                    Log.d("Note$filename", noteString!!)
+                    close()
                 }
+            }
+            if(path != null) {
                 val file = path.toFile()
-                val noteFile = File(file, "$FILENAME$numberOfFiles.$NOTE_EXT")
-                if(!noteFile.exists()) {
+                val noteFile = File(file, filename)
+                if (!noteFile.exists()) {
                     try {
                         val fileWriter = FileWriter(noteFile)
                         val bufferedWriter = BufferedWriter(fileWriter)
-                        with(bufferedWriter){
-                            write(noteString)
-                            Log.d("Note $numberOfFiles", noteString!!)
-                            close()
-                        }
+                        writeToFile(bufferedWriter)
                     } catch (e: Exception) {
                         e.printStackTrace()
                     }
                 }
             }
         }
 
+        private fun removeFileFromMemory(path: Path?){
+            if(path != null)
+                Files.deleteIfExists(path)
+        }
+
         fun saveNote(context: Context, note: TextNote?){
             if(note != null){
                 val path = getPath(context)
                 if(path != null){
-                    createFile(path,
-                        JsonParser.parseNoteToJson(note)
+                    val filename = "${sha512(note.timeStamp)}.$NOTE_EXT"
```

Comment:
Mm, no, that's something wrong here:
1) We are taking hash function of timestamp, not from resource content. This doesn't make sense then we could just use timestamp, it's small enough.
2) Sha512 is alright, but in order to integrate Memo with other ARK apps and arklib we'll need to use crc32 of resource content.

We have such a method in arklib, here is the usage: https://github.com/ARK-Builders/ARK-Navigator/blob/b8c6667b6c17b577f4710ad1f24bc93baf00f845/app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/ResourceMeta.kt#L30 You can import this dependency https://github.com/ARK-Builders/arklib-android/

Reply:
Alright, I will check that out